Mathlib/Analysis/Meromorphic/Divisor/Basic.lean

```diff
@@ -334,73 +262,51 @@ open Classical in
-subset `V` does not change its values on `V`. -/
-@[simp]
 lemma restrict_eqOn {V : Set 𝕜} (D : DivisorOn U) (h : V ⊆ U) :
     Set.EqOn (D.restrict h) D V := by
   intro _ _
   simp_all [restrict_apply, dite_eq_ite, ite_eq_left_iff]
 
-/-- Helper lemma for the `simp` tactic: restricting a divisor from `U` to a
-subset `V` makes it zero outside of `V`. -/
-@[simp]
 lemma restrict_eqOn_compl {V : Set 𝕜} (D : DivisorOn U) (h : V ⊆ U) :
     Set.EqOn (D.restrict h) 0 Vᶜ := by
   intro _ hx
   simp_all [restrict_apply, dite_eq_ite, ite_eq_left_iff, hx]
 
-/-- Restriction as an order-preserving morphism -/
-noncomputable def restrict_orderHom {V : Set 𝕜} (h : V ⊆ U) : DivisorOn U →o DivisorOn V where
-  toFun := fun D ↦ D.restrict h
-  monotone' := by
-    intro D₁ D₂ h₁₂ x
-    simp only [restrict_apply]
-    split
-    · apply h₁₂
-    · exact le_rfl
-
-/-- Helper lemma for the `simp` tactic: `restrict_orderHom` restricts divisors. -/
-@[simp]
-lemma restrict_orderHom_fun {V : Set 𝕜} (D : DivisorOn U) (h : V ⊆ U) :
-    restrict_orderHom h D = D.restrict h := by rfl
-
 /-- Restriction as a group morphism -/
-noncomputable def restrict_monoidHom {V : Set 𝕜} (h : V ⊆ U) : DivisorOn U →+ DivisorOn V where
-  toFun := fun D ↦ D.restrict h
+noncomputable def restrictMonoidHom {V : Set 𝕜} (h : V ⊆ U) : DivisorOn U →+ DivisorOn V where
+  toFun D := D.restrict h
   map_zero' := by
     ext x
     simp [restrict_apply]
-  map_add' := by
-    intro D₁ D₂
+  map_add' D₁ D₂ := by
     ext x
     by_cases hx : x ∈ V
     <;> simp [restrict_apply, hx]
 
-/-- Helper lemma for the `simp` tactic: `restrict_groupHom` restricts divisors. -/
 @[simp]
-lemma restrict_groupHom_fun {V : Set 𝕜} (D : DivisorOn U) (h : V ⊆ U) :
-    restrict_monoidHom h D = D.restrict h := by rfl
+lemma restrictMonoidHom_apply {V : Set 𝕜} (D : DivisorOn U) (h : V ⊆ U) :
+    restrictMonoidHom h D = D.restrict h := by rfl
 
 /-- Restriction as a lattice morphism -/
-noncomputable def restrict_latticeHom {V : Set 𝕜} (h : V ⊆ U) :
+noncomputable def restrictLatticeHom {V : Set 𝕜} (h : V ⊆ U) :
     LatticeHom (DivisorOn U) (DivisorOn V) where
-  toFun := fun D ↦ D.restrict h
-  map_sup' := by
-    intro D₁ D₂
+  toFun D := D.restrict h
+  map_sup' D₁ D₂ := by
     ext x
     by_cases hx : x ∈ V
     <;> simp [DivisorOn.restrict_apply, hx]
-  map_inf' := by
-    intro D₁ D₂
+  map_inf' D₁ D₂ := by
     ext x
     by_cases hx : x ∈ V
     <;> simp [DivisorOn.restrict_apply, hx]
 
-/-- Helper lemma for the `simp` tactic: `restrict_latticeHom` restricts divisors. -/
 @[simp]
-lemma restrict_latticeHom_fun {V : Set 𝕜} (D : DivisorOn U) (h : V ⊆ U) :
-    restrict_latticeHom h D = D.restrict h := by rfl
+lemma restrictLatticeHom_apply {V : Set 𝕜} (D : DivisorOn U) (h : V ⊆ U) :
+    restrictLatticeHom h D = D.restrict h := by rfl
+
+/-- Restriction as an order-preserving morphism -/
+noncomputable example {V : Set 𝕜} (h : V ⊆ U) : DivisorOn U →o DivisorOn V :=
```

Comment:
This example is there to illustrate that you only need the lattice hom. The example can be removed from the final PR.

Reply:
Roger.